Random IOExceptions in 1.3.0

This  started happening with 1.3.0-RC5 and occasionally happens in 1.3.0:

```
[error] Could not create file /Users/user/work/proj/target/streams/_global/ivyConfiguration/_global/streams/outjava.io.IOException: No such file or directory
[error] 	at sbt.internal.io.ErrorHandling$.translate(ErrorHandling.scala:20)
[error] 	at sbt.io.IO$.$anonfun$touch$2(IO.scala:315)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at sbt.internal.io.Retry$.liftedTree2$1(Retry.scala:38)
[error] 	at sbt.internal.io.Retry$.impl$1(Retry.scala:38)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:52)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:24)
[error] 	at sbt.io.IO$.touch(IO.scala:312)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.make(Streams.scala:187)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.text(Streams.scala:161)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.log(Streams.scala:182)
[error] 	at sbt.std.TaskStreams.log(Streams.scala:70)
[error] 	at sbt.std.TaskStreams.log$(Streams.scala:70)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.log$lzycompute(Streams.scala:141)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.log(Streams.scala:141)
[error] 	at sbt.Classpaths$.$anonfun$mkIvyConfiguration$1(Defaults.scala:3218)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:280)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] 	at sbt.Execute.work(Execute.scala:289)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:280)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:834)
[error] Caused by: java.io.IOException: No such file or directory
[error] 	at java.base/java.io.UnixFileSystem.createFileExclusively(Native Method)
[error] 	at java.base/java.io.File.createNewFile(File.java:1024)
[error] 	at sbt.io.IO$.$anonfun$touch$4(IO.scala:315)
[error] 	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
[error] 	at sbt.internal.io.ErrorHandling$.translate(ErrorHandling.scala:18)
[error] 	at sbt.io.IO$.$anonfun$touch$2(IO.scala:315)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at sbt.internal.io.Retry$.liftedTree2$1(Retry.scala:38)
[error] 	at sbt.internal.io.Retry$.impl$1(Retry.scala:38)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:52)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:24)
[error] 	at sbt.io.IO$.touch(IO.scala:312)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.make(Streams.scala:187)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.text(Streams.scala:161)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.log(Streams.scala:182)
[error] 	at sbt.std.TaskStreams.log(Streams.scala:70)
[error] 	at sbt.std.TaskStreams.log$(Streams.scala:70)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.log$lzycompute(Streams.scala:141)
[error] 	at sbt.std.Streams$$anon$2$$anon$3.log(Streams.scala:141)
[error] 	at sbt.Classpaths$.$anonfun$mkIvyConfiguration$1(Defaults.scala:3218)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:280)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] 	at sbt.Execute.work(Execute.scala:289)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:280)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:834)
[error] (distage / ivyConfiguration) Could not create file /Users/user/work/proj/target/streams/_global/ivyConfiguration/_global/streams/outjava.io.IOException: No such file or directory```

I have no repro, and the probability of such an error is very low. I'm using MacOS if that matters.

--- a/main/src/main/scala/sbt/Defaults.scala
+++ b/main/src/main/scala/sbt/Defaults.scala
@@ -484,10 +484,8 @@ object Defaults extends BuildCommon {
       sbtPlugin.value,
       crossPaths.value
     ),
-    clean := {
-      val _ = clean.value
-      IvyActions.cleanCachedResolutionCache(ivyModule.value, streams.value.log)
-    },
+    cleanIvy := IvyActions.cleanCachedResolutionCache(ivyModule.value, streams.value.log),
+    clean := clean.dependsOn(cleanIvy).value,
     scalaCompilerBridgeBinaryJar := None,
     scalaCompilerBridgeSource := ZincLmUtil.getDefaultBridgeModule(scalaVersion.value),
   )

--- a/main/src/main/scala/sbt/Keys.scala
+++ b/main/src/main/scala/sbt/Keys.scala
@@ -195,6 +195,7 @@ object Keys {
   val semanticdbOptions = settingKey[Seq[String]]("The Scalac options introduced for SemanticDB").withRank(CSetting)
 
   val clean = taskKey[Unit]("Deletes files produced by the build, such as generated sources, compiled classes, and task caches.").withRank(APlusTask)
+  private[sbt] val cleanIvy = taskKey[Unit]("Deletes the ivy cached resolution")
   val console = taskKey[Unit]("Starts the Scala interpreter with the project classes on the classpath.").withRank(APlusTask)
   val consoleQuick = TaskKey[Unit]("consoleQuick", "Starts the Scala interpreter with the project dependencies on the classpath.", ATask, console)
   val consoleProject = taskKey[Unit]("Starts the Scala interpreter with the sbt and the build definition on the classpath and useful imports.").withRank(AMinusTask)

